### PR TITLE
remove support for old-style Tower Auth Tokens in 3.3

### PR DESCRIPTION
**Pull request by ryanpetrello**
related: https://github.com/ansible/tower-cli/issues/478

### CODE REVIEW

File: tower_cli/api.py [hunk 1]
```diff
@@ -36,7 +36,23 @@
...
+        if settings.use_token:
+            resp = cli_client._make_request(
+                'OPTIONS', cli_client.prefix + 'authtoken/', [], {}
+            )
+            if not resp.ok:
+                secho(
+                    'use_token is not supported in this version of Tower '
+                    '(the Auth Token API has been replaced with OAuth2.0 support)',
+                    fg='yellow', bold=True
+                )
```
Comment (AlanCoding) [inline tower_cli/api.py]: I don't feel great about emitting this as `secho`, as opposed to a warning. Example:

https://github.com/ansible/tower-cli/blob/499f4ad39c92f1b619c042ecdf0c559693c7b92d/tower_cli/cli/misc.py#L106-L108

I think the default behavior should still be to show it on every command.

**Comment by ryanpetrello** [comment]
@rooftopcellist yep, this PR is just a precursor to actual OAuth2.0 support.

**Comment by AlanCoding** [comment]
wait, tests!!

**Comment by ryanpetrello** [comment]
https://travis-ci.org/ansible/tower-cli/jobs/347369674

O_o this definitely passes for me locally...

**Pull request merged**